### PR TITLE
Add a spinner for dialog field refreshes

### PR DESCRIPTION
**Pull request by eclarizio**
This simply shows/hides a patternfly icon while the ajax transaction for refreshing a dialog field is happening so that the user can tell something is going on.

### CODE REVIEW

**Comment by dclarizio** [comment]
@himdel please review

File: client/app/services/dialog-field-refresh.service.js [hunk 1]
```diff
@@ -24,6 +24,14 @@
           }
         });
 
+        angular.forEach(allDialogFields, function(dialogField) {
+          angular.forEach(dialogFieldsToRefresh, function(fieldToRefreshName) {
+            if (dialogField.name === fieldToRefreshName) {
+              dialogField.beingRefreshed = true;
+            }
+          });
+        });
```
Comment (himdel) [inline client/app/services/dialog-field-refresh.service.js]: Maybe we don't have to go through all the fields, and use something like `lodash.find(allDialogFields, { name: fieldToRefreshName })` ?

**Comment by himdel** [comment]
I think it looks good :) but @eclarizio I'd rather you replaced that double nested foreach with only one foreach + lodash.find ..

**Comment by eclarizio** [comment]
@himdel Tried using a `lodash.find` but it didn't exist. Looking at the documentation, it was using `_.find`, so I used that with an `_.includes`. Didn't know we had access to those lodash functions, good call.

**Comment by himdel** [comment]
Hehe, yeah we have lodash in both ssui and the ops ui, really useful sometimes :)

As for `lodash` vs `_` .. well, at some point I want to make SSUI use `_` for consistency, so.. no problem there.. But how it's done right now in SSUI is that you indeed use `lodash` but need to inject it first..

(As in `function DialogFieldRefreshFactory(CollectionsApi, Notifications, lodash) {`)

File: client/app/services/dialog-field-refresh.service.js [hunk 2]
```diff
@@ -24,6 +24,12 @@
           }
         });
 
+        _.find(allDialogFields, function(dialogField) {
```
Comment (himdel) [inline client/app/services/dialog-field-refresh.service.js]: Wait, this is wrong.. `find` is supposed to return the first item for which the callback returns true..

You sure you didn't mean something like.. ?

```js
dialogFieldsToRefresh.forEach(function(name) {
  _.find(allDialogFields, { name: name }).beingRefreshed = true;
});
```

Comment (eclarizio) [inline client/app/services/dialog-field-refresh.service.js]: Ah, well, I don't actually care about the results of the find, so this actually still works in the UI. The anonymous function that's setting the `beingRefreshed` property is all that matters, though I agree this does look a little odd and should probably be more like your second one.

**Pull request merged**